app/javascript/components/GooglePayButton.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import React, { Component } from 'react';
+import googlePayment from 'braintree-web/google-payment';
+import { connect } from 'react-redux';
+
+import type { AppState, PaymentMethod } from '../state';
+
+type Props = {
+  client: ?any,
+  onSubmit: () => void,
```

Comment:
These are the typings for this component's props. I'm combining props that you're supposed to pass explicitly to the component: `client` which is optional and an object, and `onSubmit` which is a callback (actually, `onSubmit` is wrong since it should accept a `result` param).

The next line says this is a union type of this type, and the implicit `mapStateToProps` type (whatever `mapStateToProps` returns).